ui/app/routes/workspace/projects/project/app/build.ts

```diff
@@ -51,6 +63,28 @@ export default class BuildDetail extends Route {
 
     result.pushedArtifact = buildFromAppRoute.pushedArtifact;
 
+    let timeline: TimelineModel = {};
+    timeline.build = {
+      sequence: buildFromAppRoute.sequence,
+      status: buildFromAppRoute.status,
+    };
+
+    if (deploymentFromAppRoute) {
+      timeline.deployment = {
+        sequence: deploymentFromAppRoute.sequence,
+        status: deploymentFromAppRoute.status,
+      };
+    }
+
+    if (releaseFromAppRoute) {
+      timeline.release = {
+        sequence: releaseFromAppRoute.sequence,
+        status: releaseFromAppRoute.status,
+      };
+    }
+
+    result.timeline = timeline;
+
```

Comment:
Note for the future: we could potentially do this cross-referencing server-side in [the UI_ methods](https://github.com/hashicorp/waypoint/blob/0ee3f72cee308cd9d3a14b5691f324bb69099571/internal/server/proto/server.proto#L369-L376).